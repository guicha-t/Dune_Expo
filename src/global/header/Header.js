import React, { Component } from 'react';
import {StyleSheet, Text, View, Menu, Platform,
  Image, Button, FlatList, TouchableOpacity, AsyncStorage, Alert, ImageBackground, AppRegistry} from 'react-native';

import Store from './../../global/store/Store'
import * as cfg from "./../../Config";

export default class Header extends Component {
  constructor(props) {
    super(props);
    this.checkIsLogToken();
  }


  async _removeItemValue(key) {
    try {
      await AsyncStorage.removeItem(key);
      return true;
    }
    catch(exception) {
      return false;
    }
  }


  checkIsLogToken = async () => {
    fetch(cfg.API_URL + '/tokens/verifyToken', {
      method: 'POST',
      Accept: 'application/json',
      headers: {
        'Content-Type': 'application/json',
        token: Store.Token,
      },
      body: JSON.stringify({
        token: Store.Token,
      }),
    }).then((response) => response.json())
    .then((responseJson) => {
      if (responseJson.response != 'Token valid') {
        this._removeItemValue("localToken")
        this._removeItemValue("localType")
        Store.setToken('');
        Store.setTypeUser('')
        Store.setIsLog(false);
        Alert.alert('Session expirée - Déconnexion');
        this.props.navigation.navigate('Start');
        }
      })
      .catch((error) => {
      console.error(error);
    });
  };


  render() {
    return (
      <View style={styles.container}>
        <View style={{flex: 0.3, justifyContent:'center', paddingLeft: 10}}>
          <TouchableOpacity onPress={() => this.props.navigation.openDrawer()}>
            <Image source={require('./../../picture/header/openDrawer.png')} style={{width:24, height:24}}/>
          </TouchableOpacity>
        </View>

        <View style={{flex: 0.3}}>
          <TouchableOpacity style={styles.containerLogo} onPress={() => this.props.navigation.navigate('Dashboard')}>
            <Image
              style={{flex: 1, height: undefined, width: undefined}}
              source={require('./../../picture/header/dunelogo.png')}
              resizeMode="contain"
              />
          </TouchableOpacity>
        </View>

        <View style={{flex: 0.3, justifyContent:'center', alignItems:'flex-end', paddingRight: 10}}>
          <TouchableOpacity onPress={() => this.props.navigation.navigate('Profil')}>
            <Image
              style={{height: 32, width: 32}}
              source={require('./../../picture/header/profil.png')}
              resizeMode="contain"
              />
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: 56,
    flexDirection: 'row',
    elevation: 5,
    backgroundColor: cfg.PRIMARY,
    borderBottomColor: 'black',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2
  },
  containerMenu: {
    paddingLeft: 10,
  },
  containerLogo: {
    flex: 1,
  },
  containerProfil: {
    justifyContent: 'flex-end',
    paddingRight: 10,
  },
});
