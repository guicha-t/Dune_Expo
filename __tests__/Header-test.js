import Header from './../src/global/header/Header';
import 'isomorphic-fetch';
import React from 'react';

import renderer from 'react-test-renderer'

jest.useFakeTimers();

test('render Header correctly', () => {

    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
});