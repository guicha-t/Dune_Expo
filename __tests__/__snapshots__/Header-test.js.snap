// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "black",
        "elevation": 5,
        "flexDirection": "row",
        "height": 56,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
      },
      Object {
        "backgroundColor": "#f6c6d2",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 0.3,
        "justifyContent": "center",
        "paddingLeft": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.4,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 0.3,
        "justifyContent": "center",
        "paddingRight": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 32,
            "width": 32,
          }
        }
      />
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderColor": "#ea4335",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginHorizontal": 30,
            "maxWidth": 350,
            "shadowColor": "#000000",
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "justifyContent": "center",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "lineHeight": 25,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              ATTENTION !
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#666",
                    "fontSize": 17,
                    "paddingTop": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Vous devez accepter les conditions d'utilisation avant de pouvoir utiliser l'application.
            </Text>
          </View>
          <View
            style={
              Object {
                "borderRadius": 5,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ea4335",
                  "borderRadius": 5,
                  "marginBottom": 20,
                  "marginHorizontal": 10,
                  "minWidth": 90,
                  "opacity": 1,
                  "paddingVertical": 8,
                }
              }
              testID="buttonConfirm"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFF",
                      "fontSize": 15,
                      "fontWeight": "600",
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                COMPRIS
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
